test/boost/sstable_compaction_test.cc

```diff
@@ -2519,7 +2519,13 @@ SEASTAR_TEST_CASE(sstable_scrub_skip_mode_test) {
             // We expect the scrub with mode=srub::mode::abort to stop on the first invalid fragment.
             sstables::compaction_type_options::scrub opts = {};
             opts.operation_mode = sstables::compaction_type_options::scrub::mode::abort;
-            compaction_manager.perform_sstable_scrub(table->as_table_state(), opts).get();
+            bool swallowed = false;
+            try {
+                compaction_manager.perform_sstable_scrub(table->as_table_state(), opts).get();
+            } catch (sstables::compaction_aborted_exception& e) {
+                swallowed = true;
+            }
+            BOOST_REQUIRE(swallowed);
```

Comment:
`BOOST_REQUIRE_THROW` is your friend here :)
(and calling the requirement `swallowed` is a bit misleading since it's swallowed by the test, not by the code you're testing, and BOOST_REQUIRE better check the code-under-test semantics, not what the test did...)